flocker/node/test/test_loop.py

```diff
@@ -1135,11 +1148,26 @@ def test_connected(self):
         """
         service = self.service
         service.cluster_status = fsm = StubFSM()
-        client = object()
+        client = build_protocol()
         service.connected(client)
         self.assertEqual(fsm.inputted,
                          [_ConnectedToControlService(client=client)])
 
+    def test_send_era_on_connect(self):
+        """
+        Upon connecting a ``SetNodeEraCommand`` is sent with the current
+        node's era and UUID.
+        """
+        protocol = AgentAMP(self.reactor, self.service)
+        locator = UpdateNodeEraLocator()
+        peer = AMP(locator=locator)
+        pump = connectedServerAndClient(lambda: protocol, lambda: peer)[2]
+        pump.flush()
+        self.assertEqual(
+            dict(era=locator.era, uuid=locator.uuid),
+            dict(era=unicode(self.service.era),
+                 uuid=unicode(self.deployer.node_uuid)))
```

Comment:
This test also took me a while to puzzle out, partly due to my unfamiliarity with this subsystem & AMP.

AIUI, `locator` represents the control service (or also another agent?), or at least is the restricted subset of what the control service is that's relevant to this test. `protocol` represents the "the current node" from the docstring.

If that's right, maybe the test would be more clear if `protocol` were renamed to `node`, the current `peer` instantiated in the lambda and `locator` renamed to `peer`. Although maybe that's glossing over important technicalities. 

Up to you what you do with this bit of confusion.